pynamodb/models.py

```diff
@@ -648,6 +648,9 @@ def query(cls,
                 limit -= 1
             yield cls.from_raw_data(item)
 
+        if limit is None and page_size is not None:
```

Comment:
What about when the total number of results matching the query is `> page_size`?

From: http://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html
> Also, if the processed data set size exceeds 1 MB before DynamoDB reaches this limit, it stops the operation and returns the matching values up to the limit